### PR TITLE
[CN-449] Add ping endpoint and restructure repo as a general purpose agent

### PR DESCRIPTION
**Pull request by kutluhanmetin**
## Description
- /dial endpoint added for WAN Replication.
- restructured as a general purpose agent.

## User Impact
When creating a WAN Replication CR, sser is going to be sure that target cluster is reachable.

### CODE REVIEW

File: init/restore/bucket_to_hostpath_cmd.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"github.com/hazelcast/platform-operator-agent/init/bucket"
```
Comment (cagric0) [inline init/restore/bucket_to_hostpath_cmd.go:7]: This import needs to be put next to the `current-project/`(line-18) imports. Could you check all the files? Our import style is like this:

Default lib
..
Other lib
..
Current Project import

File: sidecar/router.go [hunk 2]
```diff
@@ -208,31 +215,50 @@ func (s *service) cancelHandler(w http.ResponseWriter, r *http.Request) {
...
+		err := tryDial(e)
+		if err != nil {
+			errStr := fmt.Errorf("%s is not reachable", e)
+			errs = append(errs, errStr)
+			log.Println(errStr)
+		}
+	}
+
+	if len(errs) > 0 {
+		serverutil.HttpJSON(w, DialResponse{Success: false})
```
Comment (cagric0) [inline sidecar/router.go:248]: We could send unreachable endpoints back to the operator, then in the wanReplication status we could show it. WDYT?

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
-VERSION ?= latest-snapshot
-IMAGE_TAG_BASE ?= hazelcast/platform-operator-agent
+VERSION ?= 0.1.14
+IMAGE_TAG_BASE ?= kutluhanhazelcast/platform-operator-agent
```
Comment (SeriyBg) [inline Makefile:2]: Leftover?

File: sidecar/router.go [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+)
+
+const (
+	DirName = "hot-backup"
+)
+
+// Service handles requests and keeps track of Tasks
+type Service struct {
+	Mu    sync.RWMutex
+	Tasks map[uuid.UUID]*task
```
Comment (SeriyBg) [inline sidecar/router.go:28]: Can a `sync.Map` be used instead of the Mutex?

Comment (kutluhanmetin) [reply sidecar/router.go:28]: Yes maybe, actually it does not related to this feature, do you still want me to change it?

File: Makefile (hunk 3 above)
Comment (kutluhanmetin) [reply Makefile]: Thanks 👍

File: sidecar/router.go (hunk 2 above)
Comment (kutluhanmetin) [reply sidecar/router.go]: Done 👍

File: sidecar/router.go (hunk 4 above)
Comment (SeriyBg) [reply sidecar/router.go:29]: I guess it's better to do it in a separate PR, if so

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**